### ISSUE TITLE
autoreconf failed: possibly undefined macro: AC_MSG_ERROR

### ISSUE THREAD

**Issue by EdXu** [facebook/proxygen#191]
configure.ac:43: error: possibly undefined macro: AC_MSG_ERROR
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
autoreconf: /usr/bin/autoconf failed with exit status: 1
[root@pingzilao proxygen]# autoreconf -ivf
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force 
autoreconf: configure.ac: tracing
autoreconf: running: libtoolize --copy --force
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, `build-aux'.
libtoolize: copying file `build-aux/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIR, `m4'.
libtoolize: copying file `m4/libtool.m4'
libtoolize: copying file `m4/ltoptions.m4'
libtoolize: copying file `m4/ltsugar.m4'
libtoolize: copying file `m4/ltversion.m4'
libtoolize: copying file `m4/lt~obsolete.m4'
libtoolize: Consider adding `-I m4' to ACLOCAL_AMFLAGS in Makefile.am.
libtoolize: `AC_PROG_RANLIB' is rendered obsolete by `LT_INIT'
autoreconf: running: /usr/bin/autoconf --force
configure.ac:43: error: possibly undefined macro: AC_MSG_ERROR
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
autoreconf: /usr/bin/autoconf failed with exit status: 1

**Comment by w-o-o** [comment]
@EdXu please provide the compiling environment, like OS version

here is some info you can refer, https://www.gnu.org/savannah-checkouts/gnu/autoconf/manual/autoconf-2.69/html_node/Printing-Messages.html

**Comment by erezzi** [comment]
I'm trying to build proxygen on CentOS 7 with the deps-centos.sh script. I had to fix a few issues in that script. Will push them upstream once I'm able to build proxygen.

The fixes so far are in the following patch file:
[centos_fixes.patch.gz](https://github.com/facebook/proxygen/files/1352248/centos_fixes.patch.gz)

I ran deps-centos.sh and failed while building folly. Then, I deleted the folly dir, followed the [instructions](https://github.com/mapd/mapd-core#centos-7)  and reran deps-centos.sh (so now it skipped the installation of glog, gflags and double-conversion). It failed again:

configure:17137: error: possibly undefined macro: AC_MSG_NOTICE
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
autoreconf: /usr/local/mapd-deps/20170926/bin/autoconf failed with exit status: 1

This time, the mapd-deps module was enabled. Not sure if that is what I was supposed to do.

CentOS version:

vagrant@localhost:proxygen/proxygen ‹master*›$ cat /etc/centos-release
CentOS Linux release 7.3.1611 (Core)

If anyone can propose a fix on top of my deps-centos patch, that would be great. Thanks.
